diff --git a/omx/gstomxvideo.c b/omx/gstomxvideo.c
index 9feda19..ffc9070 100644
--- a/omx/gstomxvideo.c
+++ b/omx/gstomxvideo.c
@@ -235,3 +235,15 @@ gst_omx_video_is_equal_framerate_q16 (OMX_U32 q16_a, OMX_U32 q16_b)
    * an unnecessary re-negotiation. */
   return fabs (((gdouble) q16_a) - ((gdouble) q16_b)) / (gdouble) q16_b < 0.01;
 }
+
+#ifdef OMX_TARGET_RCAR_RZV2M
+void gst_omx_video_calc_power (int base, int val, int * power)
+{
+  *power = base;
+  while (val > *power)
+  {
+    *power *= base;
+  }
+  return;
+}
+#endif
diff --git a/omx/gstomxvideo.h b/omx/gstomxvideo.h
index 9251547..5d54964 100644
--- a/omx/gstomxvideo.h
+++ b/omx/gstomxvideo.h
@@ -63,6 +63,10 @@ OMX_U32 gst_omx_video_calculate_framerate_q16 (GstVideoInfo * info);
 
 gboolean gst_omx_video_is_equal_framerate_q16 (OMX_U32 q16_a, OMX_U32 q16_b);
 
+#ifdef OMX_TARGET_RCAR_RZV2M
+void gst_omx_video_calc_power (int base, int val, int * power);
+#endif
+
 G_END_DECLS
 
 #endif /* __GST_OMX_VIDEO_H__ */
-- 
2.7.4

