diff --git a/omx/gstomxh265dec.c b/omx/gstomxh265dec.c
index 307a6e5..2c0aa87 100644
--- a/omx/gstomxh265dec.c
+++ b/omx/gstomxh265dec.c
@@ -216,9 +216,11 @@ gst_omx_h265_dec_set_format (GstOMXVideoDec * dec, GstOMXPort * port,
   return TRUE;
 #endif
 
+#ifndef OMX_TARGET_RCAR_RZV2M
 #ifdef HAVE_H265DEC_EXT
   port_def.format.video.eCompressionFormat = OMXR_MC_VIDEO_CodingHEVC;
 #endif
+#endif
   err = gst_omx_port_update_port_definition (port, &port_def);
   if (err != OMX_ErrorNone)
     return FALSE;
-- 
2.7.4

