diff -ur a/omx/gstomxh264utils.c b/omx/gstomxh264utils.c
--- a/omx/gstomxh264utils.c	2022-09-16 13:34:11.633717230 +0900
+++ b/omx/gstomxh264utils.c	2022-09-16 13:51:37.149695895 +0900
@@ -32,6 +32,13 @@
 
 static const H264ProfileMapping h264_profiles[] = {
   {"baseline", OMX_VIDEO_AVCProfileBaseline},
+#ifdef OMX_TARGET_RCAR_RZV2M
+  {"constrained-baseline", OMX_VIDEO_AVCProfileConstrainedBaseline},
+  {"main", OMX_VIDEO_AVCProfileMain},
+  {"high", OMX_VIDEO_AVCProfileHigh},
+  {"progressive-high", OMX_VIDEO_AVCProfileProgressiveHigh},
+  {"constrained-high", OMX_VIDEO_AVCProfileConstrainedHigh},
+#else
 #ifdef USE_OMX_TARGET_ZYNQ_USCALE_PLUS
   {"constrained-baseline",
       (OMX_VIDEO_AVCPROFILETYPE) OMX_ALG_VIDEO_AVCProfileConstrainedBaseline},
@@ -52,6 +59,7 @@
   {"high-4:2:2-intra",
       (OMX_VIDEO_AVCPROFILETYPE) OMX_ALG_VIDEO_AVCProfileHigh422_Intra},
 #endif
+#endif
 };
 
 OMX_VIDEO_AVCPROFILETYPE
diff -ur a/omx/gstomxh265utils.c b/omx/gstomxh265utils.c
--- a/omx/gstomxh265utils.c	2022-09-16 13:34:11.649717229 +0900
+++ b/omx/gstomxh265utils.c	2022-09-16 20:23:55.321215581 +0900
@@ -33,6 +33,7 @@
 
 static const H265ProfileMapping h265_profiles[] = {
   {"main", OMX_VIDEO_HEVCProfileMain},
+#ifndef OMX_TARGET_RCAR_RZV2M
   {"main-10", OMX_VIDEO_HEVCProfileMain10},
 #ifdef USE_OMX_TARGET_ZYNQ_USCALE_PLUS
   {"main-still-picture",
@@ -57,6 +58,7 @@
   {"main-422-10-intra",
       (OMX_VIDEO_HEVCPROFILETYPE) OMX_ALG_VIDEO_HEVCProfileMain422_10_Intra},
 #endif
+#endif
 };
 
 OMX_VIDEO_HEVCPROFILETYPE
@@ -115,6 +117,7 @@
       return OMX_VIDEO_HEVCMainTierLevel61;
     else if (g_str_equal (level, "6.2"))
       return OMX_VIDEO_HEVCMainTierLevel62;
+#ifndef OMX_TARGET_RCAR_RZV2M
   } else if (g_str_equal (tier, "high")) {
     if (g_str_equal (level, "4"))
       return OMX_VIDEO_HEVCHighTierLevel4;
@@ -132,6 +135,7 @@
       return OMX_VIDEO_HEVCHighTierLevel61;
     else if (g_str_equal (level, "6.2"))
       return OMX_VIDEO_HEVCHighTierLevel62;
+#endif
   }
 
   return OMX_VIDEO_HEVCLevelUnknown;
