--- a/omx/gstomxvideodec.c	2022-10-28 14:19:38.000000000 +0900
+++ b/omx/gstomxvideodec.c	2022-10-28 15:55:36.335495635 +0900
@@ -3392,6 +3392,7 @@
         guint8 *next_nal = NULL;
         guint8 *startcode = NULL;
         guint8 *data_end = NULL;
+        guint32 tmp = 0;
 
         if (!gst_buffer_map (frame->input_buffer, &map, GST_MAP_READ)) {
           GST_ERROR_OBJECT (self, "Failed to create a GstBuffer mapping");
@@ -3421,8 +3422,9 @@
                 offset, startcode_len);
               startcode = gst_omx_video_dec_get_startcode_length(next_nal, data_end, &startcode_len);
               if (NULL != startcode) {
-                  offset += startcode - &map.data[offset];
-                  buf->omx_buf->nFilledLen -= startcode - &map.data[offset];
+                  tmp = startcode - &map.data[offset];
+                  offset += tmp;
+                  buf->omx_buf->nFilledLen -= tmp;
               } else {
                   offset = buf->omx_buf->nFilledLen;
                   buf->omx_buf->nFilledLen = 0;
@@ -3491,8 +3493,9 @@
                 offset, startcode_len);
               startcode = gst_omx_video_dec_get_startcode_length(next_nal, data_end, &startcode_len);
               if (NULL != startcode) {
-                  offset += startcode - &map.data[offset];
-                  buf->omx_buf->nFilledLen -= startcode - &map.data[offset];
+                  tmp = startcode - &map.data[offset];
+                  offset += tmp;
+                  buf->omx_buf->nFilledLen -= tmp;
               } else {
                   offset = buf->omx_buf->nFilledLen;
                   buf->omx_buf->nFilledLen = 0;
