diff -ur a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
--- a/omx/gstomxvideodec.c	2022-11-02 16:18:39.642652115 +0900
+++ b/omx/gstomxvideodec.c	2022-11-02 20:28:59.634345621 +0900
@@ -2990,6 +2990,8 @@
 #ifdef OMX_TARGET_RCAR_RZV2M
   { /* update output resolution */
       OMX_PARAM_PORTDEFINITIONTYPE out_port_def;
+
+      gst_omx_port_get_port_definition(self->dec_out_port, &out_port_def);
       out_port_def.format.video.nFrameWidth = info->width;
       out_port_def.format.video.nFrameHeight = info->height;
       out_port_def.format.video.nStride = info->width;
