--- a/omx/gstomxvideodec.c	2022-12-27 16:21:24.000000000 +0900
+++ /bgst/gstomxvideodec.c	2022-12-27 16:23:46.000000000 +0900
@@ -3078,10 +3078,9 @@ gst_omx_video_dec_set_format (GstVideoDe
       out_port_def.format.video.nSliceHeight = GST_ROUND_UP_32 (info->height / 2) * 2; // align 32
 
       out_port_def.nBufferSize = (out_port_def.format.video.nStride * out_port_def.format.video.nSliceHeight * 3) / 2;
-
       if (out_port_def.nBufferSize <= OUT_BUFFER_LIMIT)
-        out_port_def.nBufferCountActual = 32;
-      else 
+          out_port_def.nBufferCountActual = 17;
+      else
         out_port_def.nBufferCountActual = 9;
 
       if (gst_omx_port_update_port_definition (self->dec_out_port,
