diff -ur a/gst/videoparsers/gsth264parse.c b/gst/videoparsers/gsth264parse.c
--- a/gst/videoparsers/gsth264parse.c	2022-03-11 19:59:24.000000000 +0900
+++ b/gst/videoparsers/gsth264parse.c	2022-11-02 16:16:25.602654851 +0900
@@ -1902,8 +1902,13 @@
     gint par_n, par_d;
 
     if (sps->frame_cropping_flag) {
+#ifdef GST_TARGET_RCAR_RZV2M
+      crop_width = sps->width;
+      crop_height = sps->height;
+#else
       crop_width = sps->crop_rect_width;
       crop_height = sps->crop_rect_height;
+#endif
     } else {
       crop_width = sps->width;
       crop_height = sps->height;
diff -ur a/gst/videoparsers/gsth265parse.c b/gst/videoparsers/gsth265parse.c
--- a/gst/videoparsers/gsth265parse.c	2022-01-07 11:13:46.000000000 +0900
+++ b/gst/videoparsers/gsth265parse.c	2022-11-02 15:31:32.062709814 +0900
@@ -1637,8 +1637,13 @@
     guint bit_depth_chroma;
 
     if (sps->conformance_window_flag) {
+#ifdef GST_TARGET_RCAR_RZV2M
+      crop_width = sps->width;
+      crop_height = sps->height;
+#else
       crop_width = sps->crop_rect_width;
       crop_height = sps->crop_rect_height;
+#endif
     } else {
       crop_width = sps->width;
       crop_height = sps->height;
diff -ur a/gst/videoparsers/meson.build b/gst/videoparsers/meson.build
--- a/gst/videoparsers/meson.build	2022-01-07 11:13:46.000000000 +0900
+++ b/gst/videoparsers/meson.build	2022-11-02 15:35:21.358705135 +0900
@@ -15,7 +15,7 @@
 
 gstvideoparsersbad = library('gstvideoparsersbad',
   vparse_sources,
-  c_args : gst_plugins_bad_args + [ '-DGST_USE_UNSTABLE_API' ],
+  c_args : gst_plugins_bad_args + [ '-DGST_USE_UNSTABLE_API' ] + ['-DGST_TARGET_RCAR_RZV2M'],
   include_directories : [configinc],
   dependencies : [gstcodecparsers_dep, gstbase_dep, gstpbutils_dep, gstvideo_dep],
   install : true,
