diff -ur a/gst-libs/gst/video/meson.build b/gst-libs/gst/video/meson.build
--- a/gst-libs/gst/video/meson.build	2019-04-19 18:16:20.000000000 +0900
+++ b/gst-libs/gst/video/meson.build	2022-11-01 16:42:29.992385987 +0900
@@ -124,7 +124,7 @@
 
 gstvideo = library('gstvideo-@0@'.format(api_version),
   video_sources, gstvideo_h, gstvideo_c, orc_c, orc_h,
-  c_args : gst_plugins_base_args + ['-DBUILDING_GST_VIDEO'],
+  c_args : gst_plugins_base_args + ['-DBUILDING_GST_VIDEO'] + ['-DGST_TARGET_RCAR_RZV2M'],
   include_directories: [configinc, libsinc],
   version : libversion,
   soversion : soversion,
diff -ur a/gst-libs/gst/video/video-info.c b/gst-libs/gst/video/video-info.c
--- a/gst-libs/gst/video/video-info.c	2019-09-22 02:52:11.000000000 +0900
+++ b/gst-libs/gst/video/video-info.c	2022-11-01 16:24:46.880407680 +0900
@@ -858,8 +858,13 @@
       break;
     case GST_VIDEO_FORMAT_I420:
     case GST_VIDEO_FORMAT_YV12:        /* same as I420, but plane 1+2 swapped */
+#ifdef GST_TARGET_RCAR_RZV2M
+      info->stride[0] = width;
+      info->stride[1] = GST_ROUND_UP_2 (width) / 2;
+#else
       info->stride[0] = GST_ROUND_UP_4 (width);
       info->stride[1] = GST_ROUND_UP_4 (GST_ROUND_UP_2 (width) / 2);
+#endif
       info->stride[2] = info->stride[1];
       info->offset[0] = 0;
       info->offset[1] = info->stride[0] * GST_ROUND_UP_2 (height);
@@ -912,7 +917,11 @@
       break;
     case GST_VIDEO_FORMAT_NV12:
     case GST_VIDEO_FORMAT_NV21:
+#ifdef GST_TARGET_RCAR_RZV2M
+      info->stride[0] = width;
+#else
       info->stride[0] = GST_ROUND_UP_4 (width);
+#endif
       info->stride[1] = info->stride[0];
       info->offset[0] = 0;
       info->offset[1] = info->stride[0] * GST_ROUND_UP_2 (height);
