diff --git a/source/1st_loader/src/evk_HWSetup.c b/source/1st_loader/src/evk_HWSetup.c
index 8ab0baa..d7470b3 100755
--- a/source/1st_loader/src/evk_HWSetup.c
+++ b/source/1st_loader/src/evk_HWSetup.c
@@ -40,6 +40,11 @@ void InitPFC(void)
 {
     /* Set Pin function to PFC unit */
     Set_Eachpin();
+
+    /* enable card power on */
+    PFC_WriteReg(PFC_P01_EN_MSK,  0x0100FEFF);
+    PFC_WriteReg(PFC_P01_GPIO_OE, 0x01000100);
+    PFC_WriteReg(PFC_P01_GPIO_DO, 0x01000100);
 }
 
 
